### PR TITLE
networkd: split networkd-link.c

### PR DESCRIPTION
**Pull request by yuwata**
Also, this contains a follow-up for 4eb086a38712ea98faf41e075b84555b11b54362.

@ssahani PTAL.

### CODE REVIEW

**Comment by ssahani** [comment]
This is nice . Can we have directories for them like protocol specific . It would be more cleaner as the features are growing everyday.
```
src/network/lldp
src/network/can
src/network/dhcp
ndisc
````
etc ?  Thanks for doing it.

File: src/network/networkd-link.c [hunk 1]
```diff
@@ -2080,7 +2080,7 @@ static int link_down_handler(sd_netlink *rtnl, sd_netlink_message *m, Link *link
         return 1;
 }
 
-int link_down(Link *link) {
+int link_down(Link *link, link_netlink_message_handler_t callback) {
         _cleanup_(sd_netlink_message_unrefp) sd_netlink_message *req = NULL;
```
Comment (ssahani) [inline src/network/networkd-link.c:1787]: But where is the use case we need another callback ?

Comment (yuwata) [reply src/network/networkd-link.c:1787]: CAN uses that. It is changed when CAN related things are moved.

**Comment by yuwata** [comment]
> Can we have directories for them like protocol specific. It would be more cleaner as the features are growing everyday.
> 
> ```
> src/network/lldp
> src/network/can
> src/network/dhcp
> ndisc
> ```
> 
> etc ?

Maybe in the future. I think it is not necessary now.

**Comment by ssahani** [comment]
Needs a rebase else looks good.

**Comment by ssahani** [comment]
I tested this now especially with ```link_configure_addrgen_mode```. Look Good to me. Thanks.

**Comment by yuwata** [comment]
How can I confirm that `IPv6Token=` works fine? Please give me a hint to implement a test case.

**Comment by ssahani** [comment]
ip route we do like ```sudo ./ip token set ::1a:2b:3c:4d/64 dev enp0s31f6```

```
[Match]
Name=enp0s31f6

[Network]
DHCP=yes
IPv6Token=:1a:2b:3c:4d
```

I am trying to test it but not able . I too debugging it .

**Comment by yuwata** [comment]
Thanks. I will also test that this weekend.

**Comment by yuwata** [comment]
@ssahani The commit for link local addressing was moved to #12537. So, now this PR should be almost trivial.

**Comment by ssahani** [comment]
Yes thanks this looks good. We will fix that in the other pr.

**Comment by yuwata** [comment]
@ssahani Thank you for the review.

**Pull request merged**